lib/parse.js

```diff
@@ -1278,6 +1278,218 @@ function parse($TEXT, options) {
         });
     };
 
+    function parameters() {
+        var start = S.token;
+        var first = true;
+        var params = [];
+
+        expect("(");
+
+        while (!is("punc", ")")) {
+            if (first) {
+                first = false;
+            } else {
+                expect(",");
+            }
+
+            var param = parameter();
+            params.push(param);
+
+            if (param instanceof AST_Expansion) {
+                break;
+            }
+        }
+
+        next();
+        return params;
+    }
+
+    function parameter(disallow_empty) {
+        var param;
+        var expand = false;
+        if (is("expand", "...")) {
+            expand = S.token;
+            next();
+        }
+        param = binding_element();
+
+        if (is("operator", "=") && !expand) {
+            next();
+            param.default = expression(false);
+        }
+
+        if (expand !== false) {
+            if (!is("punc", ")")) {
+                unexpected();
+            }
+            param = new AST_Expansion({
+                start: expand,
+                expression: param,
+                end: expand
+            });
+        }
+
+        return param;
+    }
+
+    function check_for_valid_parameter_identifier(identifier) {
+        switch (identifier) {
+            case "arguments":
+            case "eval":
+            case "yield":
+                if (S.input.has_directive("use strict")) {
+                    token_error(S.token, "SyntaxError: Unexpected " + identifier + " identifier as parameter inside strict mode");
```

Comment:
formatting: two space indent for `case`, and two space indent for statements following `case`